op-node/benchmarks/batchbuilding_test.go

```diff
@@ -0,0 +1,128 @@
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-service/testutils"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+func RandomSingularBatch(rng *rand.Rand, txCount int, chainID *big.Int) *derive.SingularBatch {
+	signer := types.NewLondonSigner(chainID)
+	baseFee := big.NewInt(rng.Int63n(300_000_000_000))
+	txsEncoded := make([]hexutil.Bytes, 0, txCount)
+	// force each tx to have equal chainID
+	for i := 0; i < txCount; i++ {
+		tx := testutils.RandomTx(rng, baseFee, signer)
+		txEncoded, err := tx.MarshalBinary()
+		if err != nil {
+			panic("tx Marshal binary" + err.Error())
+		}
+		txsEncoded = append(txsEncoded, hexutil.Bytes(txEncoded))
+	}
+	return &derive.SingularBatch{
+		ParentHash:   testutils.RandomHash(rng),
+		EpochNum:     rollup.Epoch(1 + rng.Int63n(100_000_000)),
+		EpochHash:    testutils.RandomHash(rng),
+		Timestamp:    uint64(rng.Int63n(2_000_000_000)),
+		Transactions: txsEncoded,
+	}
+}
+
+type BatchingBenchmarkTC struct {
+	BatchType  uint
+	BatchCount int
+	txPerBatch int
+}
+
+func (t BatchingBenchmarkTC) String(cName string) string {
+	var btype string
+	if t.BatchType == derive.SingularBatchType {
+		btype = "Singular"
+	}
+	if t.BatchType == derive.SpanBatchType {
+		btype = "Span"
+	}
+	return fmt.Sprintf("BatchType=%s, txPerBatch=%d, BatchCount=%d, Compressor=%s", btype, t.txPerBatch, t.BatchCount, cName)
+}
+
+// BenchmarkChannelOut benchmarks the performance of adding singular batches to a channel out
+// this exercises the compression and batching logic, as well as any batch-building logic
+// Every Compressor in the compressor map is benchmarked for each test case
+func BenchmarkChannelOut(b *testing.B) {
+	rc, _ := compressor.NewRatioCompressor(compressor.Config{
+		TargetFrameSize:  100000,
+		TargetNumFrames:  1,
+		ApproxComprRatio: 0.4,
+	})
+	sc, _ := compressor.NewShadowCompressor(compressor.Config{
+		TargetFrameSize:  100000,
+		TargetNumFrames:  1,
+		ApproxComprRatio: 0.4,
+	})
+	nc, _ := compressor.NewNonCompressor(compressor.Config{
+		TargetFrameSize:  100000,
+		TargetNumFrames:  1,
+		ApproxComprRatio: 0.4,
+	})
+
+	compressors := map[string]derive.Compressor{
+		"RatioCompressor":  rc,
+		"ShadowCompressor": sc,
+		"NonCompressor":    nc,
+	}
+
+	tests := []BatchingBenchmarkTC{
+		// Singular Batch Tests
+		// low-throughput chains
+		{derive.SingularBatchType, 10, 1},
+		{derive.SingularBatchType, 50, 1},
+		{derive.SingularBatchType, 100, 1},
+		{derive.SingularBatchType, 200, 1},
+		{derive.SingularBatchType, 1000, 1},
+
+		// higher-throughput chains
+		{derive.SingularBatchType, 10, 10},
+		{derive.SingularBatchType, 100, 10},
+
+		// Span Batch Tests
+		// low-throughput chains
+		{derive.SpanBatchType, 10, 1},
+		{derive.SpanBatchType, 50, 1},
+		{derive.SpanBatchType, 100, 1},
+		{derive.SpanBatchType, 200, 1},
+		{derive.SpanBatchType, 1000, 1},
```

Comment:
let's also add 10k

Reply:
FYI the 10k's are commented out now because they take an *exceedingly* long to run when comparing the different compressors over a handful of iterations. I reduced the default scope of this benchmark to more quickly give results and also to not hit any RLP limits during testing.